### PR TITLE
HBASE-23288 - Backport HBASE-23251 (Add Column Family and Table Names…

### PR DESCRIPTION
**Pull request by gjacoby126**
… to HFileContext) to branch-1

### CODE REVIEW

**Comment by gjacoby126** [comment]
@apurtell @openinx @xcangCRM - this is the branch-1 backport of HBASE-23251, which you reviewed last week.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  16m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 24s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 56s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  compile  |   1m  8s |  branch-1 passed with JDK v1.7.0_242  |
| +1 :green_heart: |  checkstyle  |   2m  2s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 51s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |   1m  4s |  branch-1 passed with JDK v1.7.0_242  |
| +0 :ok: |  spotbugs  |   2m 39s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 45s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 58s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javac  |   0m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  8s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  javac  |   1m  8s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 29s |  hbase-common: The patch generated 2 new + 4 unchanged - 0 fixed = 6 total (was 4)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m  7s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |   1m  5s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  findbugs  |   4m  2s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 47s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 156m 20s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 56s |  The patch does not generate ASF License warnings.  |
|  |   | 221m 44s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.security.visibility.TestVisibilityLabelsWithACL |
|   | hadoop.hbase.client.TestAdmin1 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/822 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f26784c47763 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-822/out/precommit/personality/provided.sh |
| git revision | branch-1 / caef9f0 |
| Default Java | 1.7.0_242 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_232 /usr/lib/jvm/zulu-7-amd64:1.7.0_242 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/1/artifact/out/diff-checkstyle-hbase-common.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/1/testReport/ |
| Max. process+thread count | 4418 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by gjacoby126** [comment]
Fixed checkstyle nits, will now take a look at the test failures.

**Comment by gjacoby126** [comment]
Both failing tests pass locally for me, and neither looks related to this particular change.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 12s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  1s |  The patch appears to include 3 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 29s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 58s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  compile  |   1m  8s |  branch-1 passed with JDK v1.7.0_242  |
| +1 :green_heart: |  checkstyle  |   2m  2s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   2m 47s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 53s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |   1m  5s |  branch-1 passed with JDK v1.7.0_242  |
| +0 :ok: |  spotbugs  |   2m 39s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 43s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  7s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  javac  |   1m  7s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m  0s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 46s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 59s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |   1m  4s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  findbugs  |   4m  2s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 42s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 153m 55s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 57s |  The patch does not generate ASF License warnings.  |
|  |   | 204m 17s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestAdmin1 |
|   | hadoop.hbase.master.normalizer.TestSimpleRegionNormalizerOnCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/822 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 06cc3fc9532d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-822/out/precommit/personality/provided.sh |
| git revision | branch-1 / caef9f0 |
| Default Java | 1.7.0_242 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_232 /usr/lib/jvm/zulu-7-amd64:1.7.0_242 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/2/testReport/ |
| Max. process+thread count | 4319 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileContext.java [hunk 1]
```diff
@@ -203,11 +218,18 @@ public String getHFileName() {
   public long heapSize() {
     long size = ClassSize.align(ClassSize.OBJECT +
         // Algorithm reference, encodingon, checksumtype, Encryption.Context reference
-        5 * ClassSize.REFERENCE +
+        7 * ClassSize.REFERENCE +
```
Comment (liuml07) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileContext.java]: Seems this is the only difference when cherry-picking? I'd rather using the same expression as HBASE-23251 if possible so future change would find comfortable eyeballs.

Comment (gjacoby126) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileContext.java]: @liuml07 - the change to heap size was required to get an existing test in TestHFileBlock to pass.

Comment (liuml07) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileContext.java]: Thanks @gjacoby126 I guess my comment is more about format for easier review and future change.
```
5 * ClassSize.REFERENCE +
2 * Bytes.SIZEOF_INT +
// usesHBaseChecksum, includesMvcc, includesTags and compressTags
4 * Bytes.SIZEOF_BOOLEAN +
Bytes.SIZEOF_LONG);
```
to
```
7 * ClassSize.REFERENCE +
2 * Bytes.SIZEOF_INT +
// usesHBaseChecksum, includesMvcc, includesTags and compressTags
4 * Bytes.SIZEOF_BOOLEAN +
2 * ClassSize.ARRAY +
Bytes.SIZEOF_LONG);
```

Instead, I think it's better to have the same expression as `master` branch.
```
// Algorithm, checksumType, encoding, Encryption.Context, hfileName reference,
5 * ClassSize.REFERENCE + 2 * Bytes.SIZEOF_INT +
// usesHBaseChecksum, includesMvcc, includesTags and compressTags
4 * Bytes.SIZEOF_BOOLEAN + Bytes.SIZEOF_LONG +
//byte[] headers for column family and table name
2 * ClassSize.ARRAY + 2 * ClassSize.REFERENCE);
```

Comment (gjacoby126) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileContext.java]: done

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 12s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 23s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 59s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  compile  |   1m  6s |  branch-1 passed with JDK v1.7.0_242  |
| +1 :green_heart: |  checkstyle  |   2m  3s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   2m 49s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |   1m  4s |  branch-1 passed with JDK v1.7.0_242  |
| +0 :ok: |  spotbugs  |   2m 45s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 49s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  1s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javac  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  9s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  javac  |   1m  9s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 59s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m  0s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |   1m  6s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  findbugs  |   4m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 34s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 151m 36s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 56s |  The patch does not generate ASF License warnings.  |
|  |   | 201m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/822 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 6d3ed446bf8a 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-822/out/precommit/personality/provided.sh |
| git revision | branch-1 / caef9f0 |
| Default Java | 1.7.0_242 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_232 /usr/lib/jvm/zulu-7-amd64:1.7.0_242 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/3/testReport/ |
| Max. process+thread count | 4321 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-822/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**